src/core/qgsmaphittest.h

```diff
@@ -34,6 +38,10 @@ class CORE_EXPORT QgsMapHitTest
     QgsMapSettings mSettings;
     HitTest mHitTest;
 
+    // optional filter expression, not owned
+    /*const*/ QgsExpression* mExpression;
+    // optional polygon, not owned
+    const QgsGeometry* mPolygon;
```

Comment:
Also here - can we have owned geometry pointer, QString instead of QgsExpression pointer?